actor/builtin/miner/miner_test.go

```diff
@@ -1178,107 +1187,145 @@ func assertSlashStatus(t *testing.T, st state.Tree, vms vm.StorageMap, minerAddr
-	pip := []byte{}
-	pip = append(pip, commP[:]...)
-	pip = append(pip, commD[:]...)
+	t.Run("PIP is invalid if miner hasn't committed sector", func(t *testing.T) {
+		vmctx, verifier, minerActor := newBuilder().
+			withMessage("verifyPieceInclusionProof").
+			withSectorSet(NewSectorSet()).
+			build()
 
-	t.Run("PIP is invalid if miner has not submitted any proofs", func(t *testing.T) {
-		err := runVerifyPIP(t, 3, commP, sectorId, pip)
+		code, err := minerActor.VerifyPieceInclusion(vmctx, th.MakeCommitment(), types.NewBytesAmount(42), 2, nil)
+		require.Error(t, err)
+		require.Equal(t, "sector not committed", err.Error())
+		require.NotEqual(t, 0, int(code), "should not produce non-zero exit code")
 
-		assert.Error(t, err)
-		assert.Equal(t, "proofs out of date", err.Error())
+		require.Nil(t, verifier.LastReceivedVerifyPieceInclusionProofRequest, "should have never called the verifier")
 	})
 
-	t.Run("After submitting a PoSt", func(t *testing.T) {
-		// submit a post
-		done := types.EmptyIntSet()
-		proof := th.MakeRandomPoStProofForTest()
-		blockheightOfPoSt := uint64(8)
-		res, err = th.CreateAndApplyTestMessage(t, st, vms, minerAddr, 0, blockheightOfPoSt, "submitPoSt", ancestors, []types.PoStProof{proof}, done)
-		assert.NoError(t, err)
-		assert.NoError(t, res.ExecutionError)
-		assert.Equal(t, uint8(0), res.Receipt.ExitCode)
-
-		t.Run("Valid PIP returns true", func(t *testing.T) {
-			err := runVerifyPIP(t, 3, commP, sectorId, pip)
-
-			assert.NoError(t, err)
-		})
-
-		t.Run("PIP is invalid if miner hasn't committed sector", func(t *testing.T) {
-			wrongSectorId := sectorId + 1
-			err := runVerifyPIP(t, 3, commP, wrongSectorId, pip)
+	t.Run("PIP is invalid if miner has not submitted any PoSt proofs", func(t *testing.T) {
+		vmctx, verifier, minerActor := newBuilder().
+			withMessage("verifyPieceInclusionProof").
+			withSectorSet(sectorSetWithOneCommitment).
+			withLastPoSt(nil).
+			build()
 
-			require.Error(t, err)
-			assert.Equal(t, "sector not committed", err.Error())
-		})
+		code, err := minerActor.VerifyPieceInclusion(vmctx, commP, types.NewBytesAmount(66), 1, []byte{42})
+		require.Error(t, err)
+		assert.Equal(t, "proofs out of date", err.Error())
+		assert.NotEqual(t, 0, int(code), "should not produce non-zero exit code")
 
-		t.Run("PIP is valid if miner's PoSts are before the end of the grace period", func(t *testing.T) {
-			blockHeight := blockheightOfPoSt + PieceInclusionGracePeriodBlocks - 1
-			err := runVerifyPIP(t, blockHeight, commP, sectorId, pip)
+		require.Nil(t, verifier.LastReceivedVerifyPieceInclusionProofRequest)
+	})
 
-			assert.NoError(t, err)
-		})
+	t.Run("PIP is invalid if miner's PoSt is too far into the past", func(t *testing.T) {
+		vmctx, verifier, minerActor := newBuilder().
+			withMessage("verifyPieceInclusionProof").
+			withSectorSet(sectorSetWithOneCommitment).
+			withLastPoSt(types.NewBlockHeight(0).Sub(types.NewBlockHeight(PieceInclusionGracePeriodBlocks * 2))).
```

Comment:
This negative block height makes me nervous.  The miner env builder should have a `withHeight` option function that sets the height and then you can set the vmctx up with a big enough block height to do this subtraction.